app/Http/Controllers/TaskController.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Helpers\TaskStatus;
+use App\Models\Task;
+use App\Models\User;
+use App\Rules\ValidTaskType;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\File;
+
+class TaskController extends Controller
+{
+    /**
+     * Show the application task dashboard.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index($activeFilter = null)
+    {
+
+        $this->authorize('update', Task::class);
+
+        $user = auth()->user();
+
+        if ($activeFilter == 'sent') {
+            $tasks = Task::where('sender_user_id', $user->id)->get()->sortBy('created_at');
+        } elseif ($activeFilter == 'archived') {
+            $tasks = Task::where('recipient_user_id', $user->id)->whereIn('status', [TaskStatus::COMPLETED->value, TaskStatus::DECLINED->value])->get()->sortBy('created_at');
+        } else {
+            $tasks = Task::where('recipient_user_id', $user->id)->where('status', TaskStatus::PENDING->value)->get()->sortBy('created_at');
+        }
+
+        return view('tasks.index', compact('tasks', 'activeFilter'));
+    }
+
+    /**
+     * Store a newly created task in storage.
+     */
+    public function store(Request $request)
```

Comment:
suggestion: fetch user through service provider.
```suggestion
    public function store(Request $request, Authenticatable $user): RedirectResponse
```